File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        # need one pointer that keeps track of unique elements, this fills array 
        # need another pointer to traverse the array in search of new elements.
        if not nums:
            return 0
        slow = 0
        for fast in range(1,len(nums)):
            if nums[fast]!=nums[slow]:
                slow+=1
                nums[slow]=nums[fast]
        return slow+1
        

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.go
func removeDuplicates(nums []int) int {
    n:=len(nums)
    if n==0{
        return 0
    }
    slow:=0
    for fast:=1;fast<n;fast++{
        if nums[fast]!=nums[slow]{
            slow++
            nums[slow]=nums[fast]
        }

    }
    return slow+1
}